### PR TITLE
decision_making_statetments / it-academy / JC1-117-22 / Victor Polonnikov / tasks 01-14

### PR DESCRIPTION
**Pull request by VictPol**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java [hunk 1]
```diff
@@ -42,8 +42,44 @@ public static void main(String[] args) {
...
+                break;
+            case (2010 % 12):
+                _zodiacYearName = "Tiger";
+                break;
+            case (2011 % 12):
+                _zodiacYearName = "Rabbit";
+                break;
+            case (2012 % 12):
+                _zodiacYearName = "Dragon";
+                break;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java:82]: DRY

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task01.java [hunk 2]
```diff
@@ -36,8 +36,8 @@ public static void main(String[] args) {
      * </ul>
      */
     static boolean isTopPlayer(int sizeTop, int positionPlayerByDamage) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return false;
+        if (positionPlayerByDamage <= sizeTop) {
+            return true;
+        } else return false;
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task01.java]: fixed to easier

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java [hunk 3]
```diff
@@ -35,8 +35,14 @@ public static void main(String[] args) {
      * </ul>
      */
     static String getGreetingByHour(int hour) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        if (hour >= 6 && hour <=11) {
+            return "Good morning";
+        } else if (hour >= 12 && hour <= 17) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java]: fixed with Magic Number Solution: named Consnants

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java [hunk 4]
```diff
@@ -35,8 +35,14 @@ public static void main(String[] args) {
      * </ul>
      */
     static String getGreetingByHour(int hour) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        if (hour >= 6 && hour <=11) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java]: fixed with Magic Number Solution: named Consnants

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java [hunk 5]
```diff
@@ -35,8 +35,14 @@ public static void main(String[] args) {
...
+            return "Good morning";
+        } else if (hour >= 12 && hour <= 17) {
+            return "Good day";
+        } else if (hour >= 18 && hour <= 23) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java]: fixed with Magic Number Solution: named Consnants

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java [hunk 6]
```diff
@@ -35,8 +35,14 @@ public static void main(String[] args) {
...
+            return "Good evening";
+        } else if (hour >= 0 && hour <= 5 || hour == 24) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java]: fixed with Magic Number Solution: named Consnants

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task03.java [hunk 7]
```diff
@@ -35,8 +35,15 @@ public static void main(String[] args) {
...
-        return null;
+        final int HUNDRED_PERCENTS = 100;
+        if ((float) currentHealthPoint / maxHealthPoint * HUNDRED_PERCENTS < 25.0f && (float) currentHealthPoint / maxHealthPoint * HUNDRED_PERCENTS > 0.0f) {
+            return "RED";
+        } else if ((float) currentHealthPoint / maxHealthPoint * HUNDRED_PERCENTS < 50.0f && (float) currentHealthPoint / maxHealthPoint * HUNDRED_PERCENTS >= 25.0f) {
+            return "ORANGE";
+        } else if ((float) currentHealthPoint / maxHealthPoint * HUNDRED_PERCENTS < 75.0f && (float) currentHealthPoint / maxHealthPoint * HUNDRED_PERCENTS >= 50.0f) {
+            return "YELLOW";
+        } else if ((float) currentHealthPoint / maxHealthPoint * HUNDRED_PERCENTS <= 100.0f && (float) currentHealthPoint / maxHealthPoint * HUNDRED_PERCENTS >= 75.0f) {
+            return "GREEN";
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task03.java]: fixed, no more repeat yourself in code

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task06.java [hunk 8]
```diff
@@ -39,8 +39,14 @@ public static void main(String[] args) {
...
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        int pointsFirstPlayer = 0;
+        int pointsSecondPlayer = 0;
+        if (tanksKilledFirstPlayer == tanksKilledSecondPlayer) {
+            return "Player 1: " + tanksKilledFirstPlayer * KILL_POINT + ". Player 2: " + tanksKilledSecondPlayer * KILL_POINT;
+        } else if (tanksKilledFirstPlayer > tanksKilledSecondPlayer) {
+            return "Player 1: " + (tanksKilledFirstPlayer * KILL_POINT + BONUS_POINT) + ". Player 2: " + tanksKilledSecondPlayer * KILL_POINT;
+        } else {
+            return "Player 1: " + tanksKilledFirstPlayer * KILL_POINT + ". Player 2: " + (tanksKilledSecondPlayer * KILL_POINT + BONUS_POINT);
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task06.java]: fixed, no more repeat yourself in code

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java [hunk 9]
```diff
@@ -38,8 +38,28 @@ public static void main(String[] args) {
      * @return итоговый урон по данному типу моба
      */
     static int getTotalDamage(int damage, String typeMob, boolean hasHolyAttribute) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0;
+        int _totalDamage;
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java]: fixed, variable renamed

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java [hunk 10]
```diff
@@ -38,8 +38,17 @@ public static void main(String[] args) {
      * @return <code>latin</code>/<code>cyrillic</code>/<code>digit</code>/<code>undefined</code>
      */
     static String getSymbolType(char symbol) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        int codeDecimal = (int) symbol;
+        String _symbolType = "undefined";
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java]: fixed, variable rename

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java [hunk 11]
```diff
@@ -38,8 +38,17 @@ public static void main(String[] args) {
...
+        if (48 <= codeDecimal && codeDecimal <= 57) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java]: fixed with Magic Number Solution: named Consnants

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java [hunk 12]
```diff
@@ -38,8 +38,17 @@ public static void main(String[] args) {
...
+            _symbolType = "digit";
+        } else if ((65 <= codeDecimal && codeDecimal <= 90) || (97 <= codeDecimal && codeDecimal <= 122)) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java]: fixed with Magic Number Solution: named Consnants

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java [hunk 13]
```diff
@@ -38,8 +38,17 @@ public static void main(String[] args) {
...
+            _symbolType = "latin";
+        } else if (1040 <= codeDecimal && codeDecimal <= 1103) {
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task11.java]: fixed with Magic Number Solution: named Consnants

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task12.java [hunk 14]
```diff
@@ -43,8 +43,48 @@ public static void main(String[] args) {
...
+            _zodiacSign = "Scorpion";
+            return _zodiacSign;
+        } else if (11.23f <= dateMonthDay && dateMonthDay <= 12.21f) {
+            _zodiacSign = "Archer";
+            return _zodiacSign;
+        } else if (12.22f <= dateMonthDay && dateMonthDay <= 12.31f) {
+            _zodiacSign = "Goat";
+            return _zodiacSign;
+        }
+        return _zodiacSign;
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task12.java]: fixed partly. Variable renamed. Kindly ask let me keep this code, I like this unstandart solution.

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task13.java [hunk 15]
```diff
@@ -32,8 +32,21 @@ public static void main(String[] args) {
      * @return номер этажа
      */
     static int getFloorNumber(int numberFloors, int numberApartmentsPerFloor, int apartmentNumber) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0;
+        int _floorNumber = 0;
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task13.java]: fixed, variable rename

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java (hunk 1 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java]: fixed, no more repeat myself in code

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task10.java [hunk 16]
```diff
@@ -35,8 +35,19 @@ public static void main(String[] args) {
      * @return номер заправки.
      */
     static int getNumberStation(int firstFuelReserves, int secondFuelReserves, int thirdFuelReserves) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return 0;
+        int _numberStation;
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task10.java]: fixed, variable rename

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java [hunk 17]
```diff
@@ -35,8 +35,25 @@ public static void main(String[] args) {
      * </ul>
      */
     static String getGreetingByHour(int hour) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        final int beginningOfMorningHour = 6;
+        final int beginningOfDayHour = 12;
+        final int beginningOfEveningHour = 18;
+        final int midnightHour = 24;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task02.java:41]: use CONSTANTS

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task06.java [hunk 18]
```diff
@@ -39,8 +39,20 @@ public static void main(String[] args) {
...
+            if (tanksKilledFirstPlayer == tanksKilledSecondPlayer) {
+                return player1 + pointsFirstPlayer + player2 + pointsSecondPlayer;
+            } else if (tanksKilledFirstPlayer > tanksKilledSecondPlayer) {
+                return player1 + (pointsFirstPlayer + BONUS_POINT) + player2 + pointsSecondPlayer;
+            } else {
+                return player1 + pointsFirstPlayer + player2 + (pointsSecondPlayer + BONUS_POINT);
+            }
+        } else {
+            return null;
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task06.java:56]: not readable

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java [hunk 19]
```diff
@@ -38,8 +38,28 @@ public static void main(String[] args) {
...
+                case "ZOMBIE":
+                    totalDamage = (int) (damage * 1.5f);
+                    break;
+                case "SAINT":
+                    totalDamage = (int) (damage / 2);
+                    break;
+                case "ANIMAL":
+                case "HUMANOID":
+                case "PLANT":
+                case "GHOST":
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task08.java:54]: this `default`

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task12.java [hunk 20]
```diff
@@ -43,8 +43,35 @@ public static void main(String[] args) {
...
+            zodiacSign = "Maiden";
+        } else if (9.22f <= dateMonthDay && dateMonthDay <= 10.22f) {
+            zodiacSign = "Scales";
+        } else if (10.23f <= dateMonthDay && dateMonthDay <= 11.22f) {
+            zodiacSign = "Scorpion";
+        } else if (11.23f <= dateMonthDay && dateMonthDay <= 12.21f) {
+            zodiacSign = "Archer";
+        } else if (12.22f <= dateMonthDay && dateMonthDay <= 12.31f) {
+            zodiacSign = "Goat";
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task12.java:74]: hard

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java [hunk 21]
```diff
@@ -42,8 +42,45 @@ public static void main(String[] args) {
...
+                zodiacYearName = "Ox";
+                break;
+            case (2010 % 12):
+                zodiacYearName = "Tiger";
+                break;
+            case (2011 % numberOfMonthsInYear):
+                zodiacYearName = "Rabbit";
+                break;
+            case (2012 % numberOfMonthsInYear):
+                zodiacYearName = "Dragon";
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statements/Task14.java:82]: DRY
hard

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**